test/opexecuter.cpp

```diff
@@ -89,17 +89,16 @@ void OpExecuter::ValidateGraph(const Graph& graph,
                                const vector<string> allowed_nodes) {
   NGRAPH_VLOG(5) << "Validate graph";
   bool found_test_op = false;
-  Node* test_op;
   for (Node* node : graph.nodes()) {
     if (node->IsSource() || node->IsSink()) {
       continue;
     } else if (node->type_string() == test_op_type_) {
       // only one node of type test_op
-      ASSERT_FALSE(found_test_op) << "Only one op of type " << test_op_type_
-                                  << " should exist in the graph. Found nodes "
-                                  << node->name() << " and " << test_op->name();
+      ASSERT_FALSE(found_test_op)
+          << "Only one op of type " << test_op_type_
+          << " should exist in the graph. Found duplicate node: "
+          << node->name();
       found_test_op = true;
-      test_op = node;
```

Comment:
Not sure why we removed the test_op?

Reply:
This was triggering a warning (of uninitialized memory read - potentially crashing in case of a test failure) which was legit. The `test_op` isn't actually used any where - no?

Reply:
It is used in the error message to give more information to the user why the graph was invalid.
Suppose you are testing "Add" op's translation. And you created a graph with 2 add ops we would print the op names for both Add ops and error out as only 1 "add" op is allowed in the graph for unit testing.
The crash should not happen because of uninitialized memory as when found_test_op is set to true, test_op is also initialized. 

It is only used for providing more information to the user.

Reply:
This is the warning I was getting:
```
/nfs/site/home/avijitch/workspace/ngraph-tf/test/opexecuter.cpp:100:80: warning: ‘test_op’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                                    << node->name() << " and " << test_op->name();
```
I guess looking at the logic of this function - it's not possible for the `test_op` remain uninitialized. As soon as the control enters into the line 98 the first time the ASSERT_FALSE won't fire (as the found_test_op is set to false) - so the error message trying to print the test_op (which is uninitialized at this time) won't trigger. The static analyzer for the C++ compiler weren't able to analyze the complex scenario - so flagged this as a potential error condition. 
Initializing the `test_op` to `nullptr` is fixes this warning.